src/indices/indexed-class.lisp

```diff
@@ -293,31 +296,53 @@ also index subclasses of the class to which the slot belongs, default is T")
 
 (defvar *indexed-class-override* nil)
 
-(defmethod slot-value-using-class :before ((class indexed-class) object slot)
+(defmethod slot-value-using-class :before ((class indexed-class) object
+                                           (slot slot-definition))
```

Comment:
So this is where things get fun. The LW slot MOP functions take a slot-name as an argument instead of a slot-definition.

closer-mop only provides a wrapper to the MOP functions that the implementation provides, it's not completely a blackbox. The way I work around this, is that I specialize on the slot type, and then for LW, I create a specialized method that takes a symbol slot and changes it to a slot-definition argument. It's not super performant but gets the job done.